src/System.Private.ServiceModel/tests/Scenarios/Contract/Fault/FaultExceptionTests.cs

```diff
@@ -0,0 +1,88 @@
+
+    [WcfFact]
+    [OuterLoop]
+    public static void TestAccessFaultContractInfo()
+    {
+        MyClientBase<IWcfService> client = null;
+        IWcfService serviceProxy = null;
+        string action = @"http://tempuri.org/IWcfService/TestFaultFaultDetailFault";
+        Type detail = typeof(FaultDetail);
+
+        try
+        {
+            // *** SETUP *** \\
+            CustomBinding customBinding = new CustomBinding();
+            customBinding.Elements.Add(new TextMessageEncodingBindingElement());
+            customBinding.Elements.Add(new HttpTransportBindingElement());
+
+            client = new MyClientBase<IWcfService>(customBinding, new EndpointAddress(Endpoints.DefaultCustomHttp_Address));
+            client.Endpoint.EndpointBehaviors.Add(new TestFaultContractInfosBehavior());
+            serviceProxy = client.ChannelFactory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            var input = new object[] { new FaultDetail(), new KnownTypeA() };
+            // Use exiting service as we only need to verify FaultContractInfo can be accessed from ClientOperation
+            serviceProxy.TestFaultWithKnownType(null, input);
+
+            //Verify
+            Assert.True(testFaultWithKnownTypeClientOp != null, "Expected testFaultWithKnownTypeClientOp is NOT null");
+            Assert.True(testFaultWithKnownTypeClientOp.FaultContractInfos.Count == 1, string.Format("expected FaultContractInfos count is 1, actual: {0}", testFaultWithKnownTypeClientOp.FaultContractInfos.Count));
+            Assert.True(testFaultWithKnownTypeClientOp.FaultContractInfos[0].Action == action, string.Format("expected FaultContractInfo Action is {0}, actual: {1}", action, testFaultWithKnownTypeClientOp.FaultContractInfos[0].Action));
+            Assert.True(testFaultWithKnownTypeClientOp.FaultContractInfos[0].Detail == detail, string.Format("expected FaultContractInfo Detail is {0}, actual: {1}", detail.ToString(), testFaultWithKnownTypeClientOp.FaultContractInfos[0].Detail.ToString()));
+
+        }
+        finally
+        {
+            // *** ENSURE CLEANUP *** \\
+            ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, (ICommunicationObject)client);
+        }
+    }
+}
+
+public class TestFaultContractInfosBehavior : IEndpointBehavior
+    {
+        public TestFaultContractInfosBehavior()
+        {
+        }
+
+        public void AddBindingParameters(ServiceEndpoint endpoint, BindingParameterCollection bindingParameters)
+        {
+        }
+
+        public void ApplyClientBehavior(ServiceEndpoint endpoint, ClientRuntime clientRuntime)
+        {
+            foreach (ClientOperation clientOperation in clientRuntime.ClientOperations)
+            {
+                if (clientOperation.Name == "TestFaultWithKnownType")
+                {
+                    FaultExceptionTests.testFaultWithKnownTypeClientOp = clientOperation;
```

Comment:
Nit  -- it would be cleaner for the behavior to keep the information in a property and have the test examine it. It's always a good idea for a test helper like this to be self-contained and not aware of the test that uses it.